railties/lib/rails/generators/rails/devcontainer/devcontainer_generator.rb

```diff
@@ -140,10 +140,16 @@ def devcontainer_db_service_yaml(**options)
         end
 
         def local_rails_mount
+          rails_dev_path = if defined? Rails::Generators::RAILS_DEV_PATH
+            Rails::Generators::RAILS_DEV_PATH
+          else
+            File.expand_path("../../../../../../..", __FILE__)
+          end
+
           {
             type: "bind",
-            source: Rails::Generators::RAILS_DEV_PATH,
-            target: Rails::Generators::RAILS_DEV_PATH
+            source: rails_dev_path,
+            target: rails_dev_path
           }
```

Comment:
`Rails::Generators::RAILS_DEV_PATH` is defined in `app_generator.rb`:

https://github.com/rails/rails/blob/c6e3336cfdec5ae9c7fb98b03d0196aa56ab30f9/railties/lib/rails/generators/rails/app/app_generator.rb#L285-L288

When executing the `devcontainer` generator alone, `app_generator.rb` is never required, so the generator errors out because it doesn't know about the constant. This is the root cause of the issue.

---

Maybe it would be better if the constant could be reused, which would also result in a shorter fix.

I suggest maybe moving the `RAILS_DEV_PATH` constant to `railties/lib/rails/generators.rb`, so both `devcontainer_generator.rb` and `app_generator.rb` have access to it.

Alternatively `require "rails/generators/rails/app/app_generator"` could be added at the top of `devcontainer_generator.rb` or at the beginning of `def local_rails_mount`, but that would result in circular requires because `app_generator.rb` also requires `devcontainer_generator.rb`. So this is probably not a good idea.

Reply:
Thanks, Jerome!  I like the `railties/lib/rails/generators.rb` definition idea. Trying it out now and seems to work, besides breaking shared `SharedGeneratorTests#test_dev_option`, which uses `Rails.root`... I think that is wrong - as far as I can tell, that's the directory of the created rails app, not the gem directory... pushing those changes up shortly.

Reply:
Yes, that's much better. updated.